src/network/networkd-brvlan.c

```diff
@@ -0,0 +1,335 @@
+        if (r < 0)
+                return log_link_error_errno(link, r, "Could not close IFLA_AF_SPEC container: %m");
+
+        /* send message to the kernel */
+        r = sd_netlink_call_async(rtnl, req, set_brvlan_handler, link, 0, NULL);
+        if (r < 0)
+                return log_link_error_errno(link, r, "Could not send rtnetlink message: %m");
+
+        return 0;
+}
+
+static inline void set_bit(unsigned nr, uint32_t *addr) {
+        if (nr < BRIDGE_VLAN_BITMAP_MAX)
+                addr[nr / 32] |= (((uint32_t) 1) << (nr % 32));
+}
+
+static inline int is_vid_valid(unsigned vid) {
+        if (vid > VLANID_MAX || vid == 0)
+                return -EINVAL;
+        return 0;
+}
+
+static int parse_vid_range(const char *rvalue, unsigned *vid, unsigned *vid_end) {
+        int r;
+        char *p;
+        char *_rvalue = NULL;
+        unsigned _vid = UINT_MAX;
+        unsigned _vid_end = UINT_MAX;
+
+        assert(rvalue);
+        assert(vid);
+        assert(vid_end);
+
+        _rvalue = strdupa(rvalue);
+        p = strchr(_rvalue, '-');
+        if (p) {
+                *p = '\0';
+                p++;
+                r = safe_atou(_rvalue, &_vid);
+                if (r < 0)
+                        return r;
+
+                r = is_vid_valid(_vid);
+                if (r < 0)
+                        return r;
+
+                r = safe_atou(p, &_vid_end);
+                if (r < 0)
+                        return r;
+
+                r = is_vid_valid(_vid_end);
+                if (r < 0)
+                        return r;
+        } else {
+                r = safe_atou(_rvalue, &_vid);
+                if (r < 0)
+                        return r;
+
+                r = is_vid_valid(_vid);
+                if (r < 0)
+                        return r;
+        }
+
+        *vid = _vid;
+        *vid_end = _vid_end;
+        return r;
+}
+
+int config_parse_brvlan_vlan(const char *unit, const char *filename,
+                             unsigned line, const char *section,
+                             unsigned section_line, const char *lvalue,
+                             int ltype, const char *rvalue, void *data,
+                             void *userdata) {
+        Network *network = userdata;
+        int r;
+        unsigned vid;
+        unsigned vid_end;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = parse_vid_range(rvalue, &vid, &vid_end);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse VLAN, ignoring: %s", rvalue);
+                return 0;
+        }
+
+        if (UINT_MAX == vid_end)
+                set_bit(vid++, network->br_vid_bitmap);
+        else {
+                if (vid >= vid_end) {
+                        log_syntax(unit, LOG_ERR, filename, line, -EINVAL, "Invalid VLAN range, ignoring %s", rvalue);
```

Comment:
naah, please log only errors you receieved from some function, and do not attach metadata to log messages that actually wasn't returned anywhere. i.e. passing "0" to log_syntax is this case is actually the right thing, as you are logging about an error you noticed yourself here, and didn't receive any error here. Let's not make up error codes unless we actually want to return them.